### PR TITLE
Build nektar in standard Spack locations

### PR DESCRIPTION
**Pull request by cmacmackin**
For some reason, the Nektar++ source was being copied to a directory in the install-tree and built there. This meant the installation was unnecessarily large (as it contained build artefacts) and also caused problems when trying to build Nektar++ using the development workflow. For that, CMake was being run in the top-level directory of the repository, before the copying took place (for some reason). This caused problems when it was rerun with a different compiler/toolchain.

### CODE REVIEW

**Comment by jwscook** [comment]
There was a good reason for this in https://github.com/ExCALIBUR-NEPTUNE/NESO-Spack/pull/1. Specifically this was a work around to:
 1. stop Nektar from building its own OpenBLAS / LAPACK in addition to whatever spack has.
 2. make sure NekPy works

Are these two reasons for the workaround now obsolete?

@will-saunders-ukaea might remember more details about https://github.com/ExCALIBUR-NEPTUNE/NESO-Spack/commit/ddc682e2e9fd0690d4b82f2aa8509806f785632c?

**Comment by will-saunders-ukaea** [comment]
1. We wanted the objects for the solvers that ship with Nektar++ so that we didn't have copies of these sources in NESO. Ideally these equation solvers would be linked into a library as part of the nektar++ build (maybe @oparry-ukaea is looking into this?). Hence the copy of the solver objects.
2. Nektar++'s cmake files (i.e. the ones we use to build against nektar++) were pointing to third party libraries that were built as part of the nektar++ build (e.g. blas) and the paths in these cmake files were to the build directory (not the install directory). Hence the build in the spack directory (I'm sure there is a better solution here we really need to get up an running at the time).
3. Similar issue with NekPy build not getting copied to install directory I think.

Hence the hopefully temporary fix to build nektar in the spack install directory.

**Comment by oparry-ukaea** [comment]
1\. was fixed by #13 for Nektar's CompressibleFlowSolver, which I think is all we're using at the moment. That PR also added a CMake macro that makes it trivial to build libraries for Nektar's other built-in solvers, if we need them.

**Comment by cmacmackin** [comment]
So @oparry-ukaea  say's we've fixed (1). I've confirmed that I fixed (2) by getting Nektar to link against spack-installed versions of those libraries, rather than building them itself (see #3). To be sure, I ran `ldd` on the Nektar shared libraries and some of the executables to confirm they weren't linking to anything in the build directory. I also confirmed that the `LinearAlgebraUnitTests-rg` executable could run.

Issue (3) still appears to be an outstanding problem. I will look into that tomorrow. Probably it does just require running an extra make command or patching the CMakeLists files.

**Comment by cmacmackin** [comment]
I've now updated the Spack package to ensure that NekPy gets installed. That should sort out all of the problems which prompted the decision to build Nektar in the install tree. @jwscook, could you please review this now (or delegate to someone to review).

File: packages/nektar/package.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import llnl.util.filesystem as fs
```
Comment (jwscook) [inline packages/nektar/package.py:6]: Just to confirm, this always come bundled with spack and we don't need any extra python config?

**Review by jwscook — COMMENTED** [review]
Looks good. there's just one action to confirm the automatic pairing of the llnl python package with spack

File: packages/nektar/package.py (hunk 1 above)
Comment (cmacmackin) [reply packages/nektar/package.py:6]: Yes. It is used extensively within Spack itself.

**Review by jwscook — APPROVED** [review]

**Pull request merged**